components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java

```diff
@@ -1026,6 +1029,19 @@ private void clearExistingTokenFromCache(OAuthTokenReqMessageContext tokenMsgCtx
                     OAuth2Util.buildScopeString(existingTokenBean.getScope()));
             OAuthUtil.clearOAuthCache(existingTokenBean.getConsumerKey(), existingTokenBean.getAuthzUser());
             OAuthUtil.clearOAuthCache(existingTokenBean);
+            if (REFRESH_TOKEN_GRANT_TYPE.equals(existingTokenBean.getGrantType())) {
+                try {
+                    OauthTokenIssuer tokenIssuer = OAuth2Util.getOAuthTokenIssuerForOAuthApp(
```

Comment:
The used method can result in a db call. This needed info should already be available through the methods that's called previously. lets find that info from those places without needing a db call.